### DISCUSSION TITLE
error when application tries to install DuckDb package using gsc

### DISCUSSION THREAD

**Discussion by polestar1988** [gramineproject/gramine#1738] [General]
I am using gsc in order to run my docker image inside sgx , in my application Some packages like duckdb are going to be installed , but I get this error:
```
Emulating a raw syscall instruction. This degrades performance, consider patching your application to use Gramine syscall API.
2024/01/30 09:06:47 ERROR Command failed with error: Error="exit status 120"
2024/01/30 09:06:47 ERROR Stderr: Error="WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0xa9a2d250>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/duckdb/\nWARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0xa9a2d910>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/duckdb/\nWARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0xa9a2e250>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/duckdb/\nWARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0xa9a2f450>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/duckdb/\nWARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0xa9a2ff50>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/duckdb/\nERROR: Could not find a version that satisfies the requirement duckdb (from versions: none)\nERROR: No matching distribution found for duckdb\n"
```
fist how could I solve the performance issue ragrading to the message :
`Emulating a raw syscall instruction. This degrades performance, consider patching your application to use Gramine syscall API.`
second why installation is not occurring , should I setup some network specifications?

**Comment by dimakuv** [comment]
You should try something like this in your manifest file:
```
sys.enable_extra_runtime_domain_names_conf = true

sgx.allowed_files = [
  "file:/etc/hosts",
]
```

**Reply by polestar1988** [reply]
@dimakuv Thanks , But how could I improve the performance as I get GatWay time out err and also message from gramine:
`Emulating a raw syscall instruction. This degrades performance, consider patching your application to use Gramine syscall API`

**Reply by dimakuv** [reply]
@polestar1988 Is this a Golang application? What's this app exactly?

The `raw syscall instruction` happens (most typically) on statically linked binaries. In particular, Golang apps are statically linked.

If you have a Golang app, then you're out of luck currently. There is no robust way to fix this.

**Reply by polestar1988** [reply]
@dimakuv Yes the application is written in golang .
The application fetches information from a python script , installs needed packages on that script and then executes the python script. 
which I get timeout error.
is the issue  related to gramine or sgx itself?

**Reply by dimakuv** [reply]
> The application fetches information from a python script , installs needed packages on that script and then executes the python script.
which I get timeout error.

A Golang application downloaded a Python script and invokes a Python interpreter to run this script? And all this is run inside Gramine-SGX?

What are you trying to do inside an SGX enclave? You seem to be running too much inside SGX, the technology is simply not advanced enough to do such things.

**Reply by polestar1988** [reply]
@dimakuv Yes , I have a connector which I want to run it totally inside SGX , because all execution parts should be secured.
Do you mean sgx technology or gramine? I mean is there any other way to so improve performance ?
What if I want to use gramine Syscalls Instead of golang syscalls? is it the proper solution? How should I use gramine syscalls in my golang application?
